packages/dds/map/src/directory.ts

```diff
@@ -1551,7 +1569,10 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
         // This should make the subdirectory structure unreachable so it can be GC'd and won't appear in snapshots
         // Might want to consider cleaning out the structure more exhaustively though?
         const successfullyRemoved = this._subdirectories.delete(subdirName);
```

Comment:
>uccessfullyRemoved = this._subdirectories.delete(subdirName); [](http://example.com/codeflow?start=15&length=61)

just so I can understand, do we need to delete if previousValue === undefined ?

Reply:
So to dispose, I need the sub directory and not its name as you have in your code above: this.disposeSubDirectoryTree(subdirName);

Reply:
That is why we are getting the directory before deleting it.